torch/fx/symbolic_trace.py

```diff
@@ -121,16 +121,20 @@ def is_leaf_module(self, m: torch.nn.Module, module_qualified_name : str) -> boo
 
     def trace(self, root: torch.nn.Module) -> Graph:
         self.root = root
+        fn = type(root).forward
         self.graph = Graph()
 
-        fn = type(root).forward
         assert isinstance(fn, FunctionType)
         co = fn.__code__
         total_args = co.co_argcount + co.co_kwonlyargcount
         names_iter = iter(co.co_varnames)
         next(names_iter)  # skip self
         args : List[Any] = [root]
-        args.extend(self._proxy_placeholder(next(names_iter)) for name in range(1, total_args))
+
+        def make_proxy_placeholder():
+            name = next(names_iter)
+            return self._proxy_placeholder(name, fn.__annotations__.get(name, None))
+        args.extend(make_proxy_placeholder() for _ in range(1, total_args))
 
         if co.co_kwonlyargcount > 0 or co.co_flags & HAS_VARSTUFF:
             if co.co_flags & inspect.CO_VARARGS:
```

Comment:
does python allow annotating `*args`? I'm not sure why one'd do that, but maybe to specify that all `*args` are the same class type or something like it. It'd require some tricks, because the type annotation needs to be applied to every unpacked element instead of the entire tuple (https://stackoverflow.com/questions/37031928/type-annotations-for-args-and-kwargs). Feel free to skip it for now and just add a todo